app/plugin.go

```diff
@@ -431,12 +604,34 @@ func (a *App) InitPlugins(pluginPath, webappPath string, supervisorOverride plug
 	}
 
 	a.RemoveConfigListener(a.PluginConfigListenerId)
-	a.PluginConfigListenerId = a.AddConfigListener(func(_, cfg *model.Config) {
+	a.PluginConfigListenerId = a.AddConfigListener(func(oldCfg *model.Config, cfg *model.Config) {
 		if a.PluginEnv == nil {
 			return
 		}
 
-		a.setPluginsActive(*cfg.PluginSettings.Enable)
+		if *oldCfg.PluginSettings.Enable != *cfg.PluginSettings.Enable {
```

Comment:
This block effectively avoids triggering activation unless the global switch gets toggled *or* the plugin enablement itself changes.